hack/generator/pkg/astmodel/arm_type.go

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"fmt"
+	"go/ast"
+)
+
+// ArmType wraps an existing type to indicate that it is an ARM targeted variation
+type ArmType struct {
+	objectType ObjectType
+}
+
+// ArmType is a Type
+var _ Type = &ArmType{}
```

Comment:
Is there a mismatch here? We are using `ArmType` as a value most places it seems yet here we assert its `ptr` impls the interface.

I think technically this is fine (actually maybe this is correct). Do any of you know if we should be asserting this matching the receiver type (which would suggest this should then be: `var _ Type = ArmType{}`), or should it always be ptr?

I observe that other types that use a value for receiver (i.e. `TypeName`) do: `var _ Type = TypeName{}`